lib/charms/opensearch/v0/helper_charm.py

```diff
@@ -228,3 +231,44 @@ def mask_sensitive_information(cmd: str) -> str:
     pattern = re.compile(r"(-tspass\s+|-kspass\s+|-storepass\s+|-new\s+|pass:)(\S+)")
 
     return re.sub(pattern, r"\1" + "xxx", cmd)
+
+
+def update_grafana_dashboards_titles(charm: "OpenSearchBaseCharm") -> None:
+    """Update the titles in the specified directory to include the charm revision."""
+    revision = get_charm_revision(charm.model.unit)
+    path = charm.charm_dir / "src/grafana_dashboards"
+
+    for dashboard_path in path.iterdir():
+        if dashboard_path.is_file() and dashboard_path.suffix == ".json":
+            try:
+                _update_dashboard_title(revision, dashboard_path)
+            except (json.JSONDecodeError, IOError) as e:
+                logger.error("Failed to process %s: %s", dashboard_path.name, str(e))
+        else:
+            logger.warning("%s is not a valid JSON file", dashboard_path.name)
+
+
+def _update_dashboard_title(revision: str, dashboard_path: Path) -> None:
+    """Update the title of a Grafana dashboard file to include the charm revision."""
+    with open(dashboard_path, "r") as file:
+        dashboard = json.load(file)
+
+    old_title = dashboard.get("title")
+    if old_title:
```

Comment:
nit:
```suggestion
    if old_title := dashboard.get("title"):
```

I believe we can always assume dashboards have a title and no need to test on it, otherwise it's a bug